### DISCUSSION TITLE
I need help creating my kata.

### DISCUSSION THREAD

**Discussion by MarioHristov** [codewars/codewars.com#2562] [Q&A] [answered]
Hello, I would really appreciate if someone can tell me what do I need so that my kata functions. I did my solution and my test and I get these error messages
src/Solution.cs(1,1): error CS8652: The feature 'top-level statements' is currently in Preview and *unsupported*. To use Preview features, use the 'preview' language version.
src/Solution.cs(1,1): error CS0106: The modifier 'public' is not valid for this item
tests/Fixture.cs(2,5): error CS8652: The feature 'top-level statements' is currently in Preview and *unsupported*. To use Preview features, use the 'preview' language version.
tests/Fixture.cs(2,5): error CS0106: The modifier 'public' is not valid for this item
Do I need references? Do the test have to be in class, namespace, have libraries? 
![image](https://user-images.githubusercontent.com/81563209/148599901-7c0317c6-8f6f-4994-b0ee-e842681f276e.png)
![image](https://user-images.githubusercontent.com/81563209/148599933-ca52cb83-15d9-4ce7-9672-cb12d331093e.png)

**Comment by hobovsky** [comment, marked as answer]
You use in your kata features from C# 9 (or 10). The highest version available on codewars is C# 8.

Additionally, i am not sure NUnit can run test cases defined as top-level functions.

Test cases need to be defined in a test class. Codewars uses NUnit to run C# tests, so I'd suggest reading up on NUnit a bit.

**Reply by MarioHristov** [reply]
Hello, thanks for the help! I figured it out on my own. I'm looking for improvement so if you can be one of my testers I would gladly appreciate that! Here is a link to the kata: https://www.codewars.com/kata/61d88f5260d73c0057f41cff
